### ISSUE TITLE
Help: Correct Aspect Ratio

### ISSUE THREAD

**Issue by cyberpunkbln** [joncampbell123/dosbox-x#1152]
Hello,

i have 2 dos games with an ugly aspect ratio bug. Pinball Dreams Deluxe and Pinball Fantasies Deluxe and Pinball Illusiions.

I use the last 0.82.19 64Bit Windows precompiled from release and 1080p LCD-Monitor. Fullresolution ist always desktop.

I want to play these both games with correct aspect ratio (full rounded sphere) in both graphic settings (low/high).

Here are some infos to this three games, with it aspect ratio problems:
[http://nerdlypleasures.blogspot.com/2014/09/classic-dos-pinball-games-mode-x-and.html](http://nerdlypleasures.blogspot.com/2014/09/classic-dos-pinball-games-mode-x-and.html)

In original Dos-Box i cannot resize the windows, when i put aspect ration here in x off and resize the window, i became the perfect aspect ratio manually (eyes to monitor:)).

Give it an DosBox-X conf for that games that make the games pixel perfect? Prefab in fullscreen and fullresolution=desktop?

Otherwise i have experimented with the window-mode, to replicate my manuell windows resizing, but there are some allgos that disturb my experiments and i have no overview no more about all the settings;).

DosBox-X seems the only port that has the abilty to make this three games playable like and vga-monitor in 1990:).

thx

PS: If DosBox-X cannot view this three games aspect ratio and pixel perfect, was there an option that:

DosBox uses an borderless fullscreen settings and try optimize the resolution in this borderless mode. In Windows mode i can resize the window to a manuell pixel perfect window? This will be very cool mode, why in window mode i have all under the window (background and so on) and with the borderless mode and perfect pixel allgo i have only the black borders that i must have for this three games.

Edit: btw how can i DosBox-X say that the automaticly double the windowresolution=original? When i use aspect ratio=true and windowresolution=original he makes me a very small but perfect view of the game, when dosbox-x can this double or more excatly, this will be great;)?

**Comment by joncampbell123** [comment]
Currently the only way to obtain pixel perfect rendition is to use output=surface and disable the "aspect ratio correction". Direct3D and OpenGL are more focused on fitting the window than pixel perfect rendering.

**Comment by cyberpunkbln** [comment]
@joncampbell123 Thx, this are the combination that i don't have tried:). Dreams and Fantasies have round spheres in the low/normal graphic setting -> 320x200. 320x350 is not pixel perfect possible. Sounds strange but aspect ration false solve some aspect ration errors:).

Hmm I wonder why other Emulators can better scale old crt/vga-resolutions. UAE can it better and 86Box can it better, but its slower:).

**Comment by Wilstorm** [comment]
I know this is an old issue but I was wondering if I could ask a quick question related to these same machines. I tried the suggested settings above but they still look slightly compressed vertically. I've spent a few days here and there messing with different options to find what I like best and was wondering if "fit to aspect ratio" is an option that can be set within the .conf file? With it on, it seems to correct the 320x350 (default resolution) and makes it look correct or at least good enough for me (vs being slightly compressed vertically). The 320x200 of course clips the right side. The other modes if they were slightly bigger would be fine but are just to small for these old peepers.
